checker/src/org/checkerframework/checker/index/samelen/SameLenVisitor.java

```diff
@@ -25,6 +25,23 @@ public SameLenVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    // This variable is used to store the name of the variable being assigned to on
+    // the left hand side of an assignment/pseudo-assignment, so that that variable
+    // can later be included in the SameLen type. This pattern is a bit uncomfortable,
+    // but there is not a clear other way to save this information.
```

Comment:
...But this is also a problem with the code as is. So in principle I don't see any reason we couldn't do it this way, but regardless I need to be able to get the canonical name of the lhs of a `VariableTree`. Do you know how to do this? @smillst